### ISSUE TITLE
Error in chapter 8 KL divergence formula

### ISSUE THREAD

**Issue by tanmey007** [nlp-with-transformers/notebooks#94]
## Information

The problem arises in chapter:

* [ ] Introduction
* [ ] Text Classification
* [ ] Transformer Anatomy
* [ ] Multilingual Named Entity Recognition
* [ ] Text Generation
* [ ] Summarization
* [ ] Question Answering
* [THis one] Making Transformers Efficient in Production
* [ ] Dealing with Few to No Labels
* [ ] Training Transformers from Scratch
* [ ] Future Directions

## Describe the bug

The formula used to calculate KL divergence is wrong

## To Reproduce

1.
2.
3.

```python

loss_fct = nn.KLDivLoss(reduction="batchmean")
loss_kd = self.args.temperature ** 2 * loss_fct(F.log_softmax(logits_stu / self.args.temperature, dim=-1),F.softmax(logits_tea / self.args.temperature, dim=-1))
```

## Expected behavior
```python

loss_fct = nn.KLDivLoss(reduction="batchmean")
loss_kd = self.args.temperature ** 2 * loss_fct(F.log_softmax(logits_stu / self.args.temperature, dim=-1),F.log_softmax(logits_tea / self.args.temperature, dim=-1))
```
Reference :https://pytorch.org/docs/1.13/generated/torch.nn.KLDivLoss.html#torch.nn.KLDivLoss

**Comment by triet1102** [comment]
Hi, you might find this answer helpful https://discuss.pytorch.org/t/kl-loss-and-log-softmax/69136/2
